### PR TITLE
Login holder

### PR DESCRIPTION
**Pull request by MarcoPolo**
## Preview
loginform

This makes the boiler plate for the login page.

Basically there is a new tab called login, it doesn't show up in the top tabs on the main part of the app.

When you go there you have a carousel, and a form container. You can put anything in the form container by registering a new path for it under login/index.js (see the `intro` component for an example).

This means that do switch between pages in the login/signup form you just have to dispatch an action to update the navState.

## Dev Stuff

I added a global function to dispatch the `devEdit` function, so know you can change the store arbitrarily from the console. You can do things like:
```
devEdit(['tabbedRouter','tabs', 'tabs:loginTab', 'uri'], Immutable.List([Immutable.Map({path: 'root'}), Immutable.Map({path: 'foo'})])))
devEdit(['tabbedRouter','activeTab'], 'tabs:moreTab')
```

@keybase/react-hackers

### CODE REVIEW

File: react-native/react/login/index.render.desktop.js [hunk 1]
```diff
@@ -1,12 +1,69 @@
...
 
 export default class LoginRender extends Component {
-  render () {
+  props: LoginRenderProps;
+
+  window: ?any;
+  originalSize: ?{width: number, height: number};
+
+  componentWillMount () {
+    if (resizeLoginForm) {
```
Comment (chrisnojima) [inline react-native/react/login/index.render.desktop.js]: This is fine for now but i'm worried about a component doing this. Might be better for nav or something higher level to handle this?

File: react-native/react/login/index.render.desktop.js [hunk 2]
```diff
@@ -1,12 +1,69 @@
...
+      this.window = remote.getCurrentWindow()
+      const [width, height] = this.window.getSize()
+      this.originalSize = {width, height}
+      this.window && this.window.setSize(styles.container.width, styles.container.height + 20 /* for frame */, true)
```
Comment (chrisnojima) [inline react-native/react/login/index.render.desktop.js]: can we use 
```
useContentSize Boolean - The width and height would be used as web page's size, which means the actual window's size will include window frame's size and be slightly larger. Default is false.
``` 
?

or at least move the 20 into the hacks section of the global-styles

File: react-native/react/util/carousel.desktop.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+          <Text style={{margin: 20}} type='Body'>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus sagittis lacus vel augue laoreet.</Text>
+      </div>
+    )
+  }
+}
+
+function PageIndicator ({selectedIndex, itemCount, style, onClickPage}: {selectedIndex: number, itemCount: number, style: Object, onClickPage: (i: number) => void}): ReactElement {
+  const marks = _.range(itemCount).map(n => {
+    const color = (n === selectedIndex) ? globalColors.white : globalColors.lightBlue
+    return (<i className='fa fa-circle' style={{color, ...styles.pageIndicator}} onClick={() => onClickPage(n)} key={n}/>)
```
Comment (chrisnojima) [inline react-native/react/util/carousel.desktop.js]: we have an icon class for this in common-adapters

File: react-native/react/util/carousel.desktop.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  render (): ReactElement {
+    const marginLeft = this.state.selectedIndex * -(this.props.itemWidth + 20)
+    const itemWidth = this.props.itemWidth
+    return (
+      <div style={{...styles.carousel, ...this.props.style}}>
+        <div style={{...globalStyles.flexBoxRow, width: itemWidth, overflow: 'visible'}}>
+          <div style={{...styles.carouselWrapper, marginLeft}}>
+            <CarouselThing style={{...styles.carouselThing, ...globalStyles.rounded, width: itemWidth}}/>
```
Comment (chrisnojima) [inline react-native/react/util/carousel.desktop.js]: not using itemCount and a loop?

File: react-native/react/util/carousel.desktop.js (hunk 3 above)
Comment (MarcoPolo) [inline react-native/react/util/carousel.desktop.js]: sorry, I looked there but must have missed it!

File: react-native/react/util/carousel.desktop.js (hunk 4 above)
Comment (MarcoPolo) [inline react-native/react/util/carousel.desktop.js]: Each thing will be different when we actually make this, so I didn't bother doing a loop that we'll probably want to change later.

Still want me to change?

Comment (chrisnojima) [inline react-native/react/util/carousel.desktop.js]: its fine to leave it but its a 2 line change right? :)
:+1:

Comment (MarcoPolo) [inline react-native/react/util/carousel.desktop.js]: happy to change! just wanted to explain my reasoning

Comment (chrisnojima) [inline react-native/react/util/carousel.desktop.js]: Cool, no biggie either way. i just don't want someone else to pick it up and hard code it

**Comment by MarcoPolo** [comment]
@chrisnojima updated!

**Comment by chrisnojima** [comment]
:+1:

**Pull request merged**